### ISSUE TITLE
ZoomEnd function outputs "Cannot read property 'x' of undefined"

### ISSUE THREAD

**Issue by ricardo-dias** [marmelab/EventDrops#186]
Hi,

I've been trying to setup eventdrops to get time range and displayed data on zoomend event, but the code from the README is producing an error:
`Uncaught TypeError: Cannot read property 'x' of undefined`

Since I'm new to D3, this is likely something silly that I'm doing.

Code is as follows:

    var eventDropsChart = d3.chart.eventDrops()
      .date(d => d.date)
      .eventLineColor(d => d.color)
      .start(new Date().setMonth(new Date().getMonth() - 3))
      .end(new Date())
      .zoomend((data) => {
        const currentScale = d3.chart.scales.x;
        const newScale = d3.event ? d3.event.transform.rescaleX(currentScale) : currentScale;
        const filteredData = data.map(dataRow => chart.visibleDataInRow(dataRow.data, newScale));

        // retrieve start and end dates
        newScale.domain()[0].toLocaleDateString('en-US');
        newScale.domain()[1].toLocaleDateString('en-US');
    });

    d3.select('#chart')
      .datum(data)
      .call(eventDropsChart);
My goal is to use the min & max events in an Ajax request, to fetch only the subset of data required.

Additionally, the 'minimum' and 'maximum' values in the 'extremum' class don't get updated upon zoom events.

Can anyone give me a hint on how to solve these problems?

Thanks in advance!

**Comment by jpetitcolas** [comment]
Hi,

It seems you use an old version of EventDrops. We [released the `1.0.0` version](https://marmelab.com/blog/2018/01/26/event-drops-1.0-is-out.html) recently, bringing a lot of improvements and bug fixes, especially for the:

> Additionally, the 'minimum' and 'maximum' values in the 'extremum' class don't get updated upon zoom events.

Trying to fetch some extra data through an AJAX call is quite interesting, even if we never tried. I would indeed suggest to do it in the `zoomend` event, something like (in new version):

``` js
const eventDrops = eventDrops({
    zoom: {
        onZoomEnd: () => {
            const [startDate, endDate] = chart.scale().domain();
            // do your AJAX request here and update your selection data
        }
    }
});
```
I'll try to give it a try on my side. Not sure if updating data currently work.

Meanwhile, can you try to display all data? EventDrops makes some optimization to get good performances even with a lot of data (we filter out of viewport data and do not display two superposed points).

**Comment by jpetitcolas** [comment]
Closing as no more feedbacks.
